connection/connection.go

```diff
@@ -61,10 +54,11 @@ func NewConnection(u url.URL,
 
 	headers := http.Header{}
 	headers.Set("Authorization", "Bearer "+token)
-	ws, _, err := dialer.Dial(u.String(), headers)
+	ws, rsp, err := dialer.Dial(u.String(), headers)
 	if err != nil {
 		return nil, err
 	}
+	defer rsp.Body.Close()
```

Comment:
I am not sure about that one. was this required by the static check? or do you think it is required?

Reply:
I was required by the static check, same as we have done [for `mender-cli`](https://github.com/mendersoftware/mender-cli/pull/139/files#diff-3d829077a7d1516e689ac2ac153f9caaf9e4a0004c4c9d059c4800ec279af26dR104).

What are you unsure about? `dialer.Dial` does an HTTP request and [returns the response](https://pkg.go.dev/github.com/gorilla/websocket#Dialer.Dial) before the upgrade to websockets. This response's body [has to be closed](https://pkg.go.dev/net/http#ReadResponse).